### PR TITLE
[flang] Fix const cast issue in FreeMemory function call in execute_command_line

### PR DESCRIPTION
**Pull request by yiwu0b11**
The FreeMemory function only accepts a void pointer, but it was being called with a const char pointer, resulting in a type-casting issue.
To address this, the const was removed, use char * instead.

### CODE REVIEW

**Comment by yi-wu-arm** [comment]
@klausler point it out in the flang slack channel:
flang/runtime/execute.cpp:202:24: error: cast from type 'const char*' to type 'void*' casts away qualifiers [-Werror=cast-qual]
  202 |     FreeMemory((void *)newCmd);                                 
      |                        ^~~~~~

**Comment by yi-wu-arm** [comment]
Build and pass all test on clang 10.0.0 and on gcc 9.4.0

File: flang/docs/Intrinsics.md [hunk 1]
```diff
@@ -847,7 +847,7 @@ used in constant expressions have currently no folding support at all.
 
 #### Usage and Info
 
-- **Standard:** Fortran 2008 and later, specified in 16.9.73
+- **Standard:** Fortran 2008 and later, specified in Section 16.9.73
```
Comment (klausler) [inline flang/docs/Intrinsics.md:850]: The standard uses the word "subclause", not "section".

**Review by klausler — APPROVED** [review]

File: flang/docs/Intrinsics.md (hunk 1 above)
Comment (yi-wu-arm) [reply flang/docs/Intrinsics.md]: done

**Pull request merged**